packages/block-library/src/image/edit.js

```diff
@@ -152,21 +155,46 @@ export function ImageEdit( {
 			additionalAttributes = { url };
 		}
 
-		// Check if the image is linked to it's media.
-		if ( linkDestination === LINK_DESTINATION_MEDIA ) {
-			// Update the media link.
-			mediaAttributes.href = media.url;
+		// Check if default link setting should be used.
+		let linkDestination = attributes.linkDestination;
+		if ( ! linkDestination ) {
+			// Use the WordPress option to determine the proper default.
+			// The constants used in Gutenberg do not match WP options so a little more complicated than ideal.
+			// TODO: fix this in a follow up PR, requires updating media-text and ui component.
+			switch ( wp.media.view.settings.defaultProps.link ) {
```

Comment:
We probably want to guard against something in this chain being `undefined` if Gutenberg is running in a context that doesn't have it. Luckily we're using Babel, so we can use the optional chaining operator, `?.` and throw on a `default` case.

Reply:
Good point, it could fatal. I was thinking of null would work fine.
Addressed in: 89bfd84